### PR TITLE
Fix/copy project wiki notifications

### PR DESCRIPTION
**Pull request by ulferts**
Fixes issues when copying a wiki along with the project:
* The author is not set to the copying user. Because of that, notifications are sent out to the original author which is not necessarily the one performing the copy project in case the user has selected to be informed of everything and also disables the "I don't want to be notified by changes that I make myself". This also in effect suggests a faulty authorship.
* The "Sent notifications while copying the project" flag is disregarded for wikis. Because the email sending is carried out delayed, setting `ActionMailer.perform_deliveries` is disregarded.

https://community.openproject.com/wp/34585

### TODO

* [x] Use the journal based notification -> mail sending also employed by work packages. This way, mails are also aggregated which reduces the amount of mails being sent. It also allows disabling emails being sent. 
* [x] Create contracts and services for `WikiPage`
* [x] Set the author of a `WikiContent` to always be the current user
* [x] Wire `WikiPage` copy service up to the project copying
* [x] Set email sending in the `WikiPage` copy service

### CODE REVIEW

File: app/services/notifications/journal_notification_service.rb [hunk 1]
```diff
@@ -30,24 +30,24 @@
...
     def call(journal, send_mails)
-      if journal.journable_type == 'WorkPackage'
-        enqueue_work_package_notification(journal, send_mails)
-      end
+      enqueue_work_package_notification(journal, send_mails) if supported?(journal)
     end
 
     private
 
     def enqueue_work_package_notification(journal, send_mails)
```
Comment (machisuji) [inline app/services/notifications/journal_notification_service.rb:39]: This this is now no longer specific to work packages we should rename this method.

**Review by machisuji — APPROVED** [review]
Code looks good. I would just request one minor change where a method should be renamed.
It's not necessary to send this through the CI again for this alone, though.

So I'm going to go ahead and merge this. The method can be renamed separately.

As far as I can tell everything seems to be working just fine with the changes but this will still require thorough QA once merged.

**Comment by machisuji** [comment]
@ulferts One more thing I noticed is that copied wiki attachments still have the old author of the copy template rather than the new author. Notification-wise this is not a problem, however, which is why I didn't let this stop this PR from being merged.

We might want to adjust this too, though.

**Comment by ulferts** [comment]
@machisuji thanks for the feedback. I renamed the method and will open a ticket for the attachment with wrong author bug.

**Pull request merged**